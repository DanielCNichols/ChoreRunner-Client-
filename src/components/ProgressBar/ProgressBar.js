import React from 'react';

export default function ProgressBar({ points }) {
  const containerStyle = {
    height: '30px',
    width: '100%',
    backgroundColor: 'none',
    border: '2px solid white',
  };

  const filledStyle = {
    height: '100%',
    width: '10%',
    backgroundColor: 'white',
    textAlign: 'right',
  };

  const labelStyle = {
    padding: 5,
    color: 'black',
    fontWeight: 'bold',
  };

  return (
    <div style={containerStyle}>
      <div style={filledStyle}>
        <span style={labelStyle}>{`1 / 10`}</span>
      </div>
    </div>
  );
}
