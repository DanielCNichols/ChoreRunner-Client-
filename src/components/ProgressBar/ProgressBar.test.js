import React from 'react';
import ReactDOM from 'react-dom';
import ProgressBar from './ProgressBar';
import renderer from 'react-test-renderer';

describe('ProgressBar component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<ProgressBar progressPoints={2} />, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer.create(<ProgressBar progressPoints={2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
