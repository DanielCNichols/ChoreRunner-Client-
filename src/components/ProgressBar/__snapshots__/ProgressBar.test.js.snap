// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component testing renders UI as expected 1`] = `
<div
  style={
    Object {
      "backgroundColor": "blue",
      "border": "2px solid white",
      "height": "30px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <span
    style={
      Object {
        "color": "#ffd900",
        "display": "inline-block",
        "fontWeight": "bold",
        "padding": "5px",
        "position": "relative",
        "textAlign": "center",
        "verticalAlign": "middle",
        "width": "100%",
        "zIndex": "2",
      }
    }
  >
    8 / 10
  </span>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "height": "100%",
        "position": "absolute",
        "textAlign": "right",
        "top": "0",
        "width": "80%",
      }
    }
  />
</div>
`;
